openmetrics_udpserver/src/http_server.rs

```diff
@@ -36,19 +37,30 @@ async fn get_metrics(
         .status(StatusCode::OK)
         .header(
             "Content-Type",
-            "application/openmetrics-text; version=1.0.0; charset=utf-8",
```

Comment:
OpenMetrics schreibt ohne Begründung: "The content type MUST be: application/openmetrics-text; version=1.0.0; charset=utf-8"

Prometheus Doc: 
OpenMetrics is the an effort to standardize metric wire formatting built off of Prometheus text format. It is possible to scrape targets and it is also available to use for federating metrics since at least v2.23.0.

Das OpenMetrics format ist eine Obermenge des Prometheus formats. OpenMetrics hat zB neue Metriktypen eingeführt, siehe Beispiel ganz unten.

Der content-type im header scheint dem Scraper zu sagen, welchen Typ er da vor sich hat. Wie genau Prometheus diese Information dann verarbeiten würde weiß ich nicht.

Die Frage ist welche Metriktypen der UDP agent herausgibt bzw. herausgeben können soll. Würde man eine Metrik im OpenMetric Format herausgeben aber den content-type "text/plain" mitschickt ist es wahrscheinlich so, dass ein Scraper wie Prometheus dann beim parsen der Metriken Probleme hat. Könnte dann so aussehen

`prometheus_1 | level=warn ts=2019-04-19T17:59:44.932649585Z caller=scrape.go:835 component="scrape manager" scrape_pool=apps target=http://127.0.0.1:20171/metrics msg="append failed" err="invalid metric type \"info\""`

wie hier: https://github.com/prometheus/client_python/issues/399

**Wahrscheinlich würde Prometheus also ohne Deklarierung bei einem echten OpenMetric Typ versagen.**

Diese Änderung sollten wir also nur einführen, wenn der UDP-agent  ausschließlich den Prometheus Typ herausgibt.

.
.
.
.
.
.



Beispiel für verschiedene Metrik typen:

Prometheus:
```
# TYPE foo_seconds_total counter
foo_seconds_total 1.0
```

OpenMetrics (including optional UNIT and _created):
```
# TYPE foo_seconds counter
# UNIT foo_seconds seconds
foo_seconds_total 1.0
foo_seconds_created 1572628096.0
# EOF
```

Reply:
Ich denke mal wir müssen das pro Accept matchen der gesendet wird. Wenn ich das richtig sehe, sendet Protmetheus ein Accept für OpenMetrics-Text wenn nichts anderes konfiguriert wurde:
```
# The protocols to negotiate during a scrape with the client.
  # Supported values (case sensitive): PrometheusProto, OpenMetricsText0.0.1,
  # OpenMetricsText1.0.0, PrometheusText0.0.4.
  # The default value changes to [ PrometheusProto, OpenMetricsText1.0.0, OpenMetricsText0.0.1, PrometheusText0.0.4 ]
  # when native_histogram feature flag is set.
  [ scrape_protocols: [[<string>](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#string), ...] | default = [ OpenMetricsText1.0.0, OpenMetricsText0.0.1, PrometheusText0.0.4 ] ]
```
https://github.com/prometheus/prometheus/blob/9a9e73b34a7d6a34083f486a66ce4243b89f45c6/scrape/scrape.go#L682-L692

Ich stelle das mal entsprechend um, müssen wir dann ausprobieren ob es entsprechend klappt.